api/v1alpha1/conversion_stash.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const conversionStashAnnotation = "azure.microsoft.com/convert-stash"
+
+func getStashedAnnotation(meta metav1.ObjectMeta, target interface{}) (bool, error) {
+	if meta.Annotations == nil {
```

Comment:
My initial thought was that this isn't necessary since json.Unmarshal does the same check internally, but I think you're right since we can give a clearer error.